### ISSUE TITLE
Add support for requestFilter function

### ISSUE THREAD

**Issue by angeloinqbyte** [pact-foundation/pact-php#164]
Need to manipulate headers like auth tokens before replaying pact requests. Saw it was implemented in pact-js. Any idea how to implement this? Would be happy to assist in implementation.

**Comment by mefellows** [comment]
Hi @angeloinqbyte this would be great and we'd be supportive of this feature.

I believe the infrastructure is already in place here in the PHP code (see https://github.com/pact-foundation/pact-message-demo and https://gist.github.com/bethesque/c858e5c15649ae525ef0cc5264b8477c for some background on pact message design), so I think you could leverage something similar for the request filter - at least, this is how it's done in Pact JS/Go.

TL;DR - Spin up a proxy server in pact-php that becomes the new verification target, which has a registered middleware function to intercept and modify the payload. The proxy then passes the request to the _actual_ target.

**Comment by cfmack** [comment]
Is there an example from Pact JS (or even better for me, Pact.Net)?    You are correct that pact-php-message set serves up a proxy.

**Comment by mefellows** [comment]
Something like this https://github.com/pact-foundation/pact-js/blob/master/examples/e2e/test/provider.spec.js#L23-L32?

I don't think Pact .NET supports it sadly.

**Comment by cfmack** [comment]
Thank you so much @siad007 .   You have been adding all kinds of PRs!  Thank you
